sql/analyzer/pushdown.go

```diff
@@ -538,6 +538,10 @@ func pushdownFiltersToTable(
 	handled := ft.HandledFilters(normalizeExpressions(ctx, tableAliases, tableFilters...))
```

Comment:
What are your thoughts on making two slices `normalizedExpr` and `denormalizedExpr` with one traversal before calling `HandledFilters`? Bridge the gap afterwards with a reverse mapping, or update the handleFilters signature -> `func HandledFilters(expr) []int`? Pushdown is the most expensive rule and I think `FixFieldIndexes` gets called a lot.

Reply:
If my performance concerns are misplaced this all LGMT thought

Reply:
Good feedback! I definitely don't wanna make our slowest analyzer rule slower! 😄  I'll push a new commit with the mapping between normalized/denormalized expressions in there in a few mins when I get it all cleaned up. 

Making that change also helped me find a bug where the filter was being applied twice in the plan because of accidentally marking the normalized filter expression as handled, instead of the denormalized version.